src/main/java/org/dataloader/DataLoaderRegistry.java

```diff
@@ -97,6 +105,15 @@ protected DataLoaderRegistry(Map<String, DataLoader<?, ?>> dataLoaders, DataLoad
         }
     }
 
+    private static DataLoader<?, ?> checkAndSetName(String key, DataLoader<?, ?> dataLoader) {
+        if (dataLoader.getName() == null) {
+            return dataLoader.transform(b -> b.name(key));
+        }
+        assertState(key.equals(dataLoader.getName()),
+                () -> String.format("Data loader name '%s' is not the same as registered key '%s'", dataLoader.getName(), key));
+        return dataLoader;
+    }
```

Comment:
It validates that a DL named "Foo" has been registered as the key "Foo" because `dlr.register("Bar", fooDL)` makes not sense